graphql/util_test.go

```diff
@@ -351,3 +353,63 @@ func TestConcurrentlyBuildVersionsMatchingTasksMap(t *testing.T) {
 	assert.Equal(t, versionsMatchingTasksMap["v3"], true)
 
 }
+
+func TestHasAnnotationPermission(t *testing.T) {
+	for tName, tCase := range map[string]func(ctx context.Context, t *testing.T){
+		"HasAnnotationPermission": func(ctx context.Context, t *testing.T) {
+			task := restModel.APITask{ProjectId: utility.ToStringPtr("project_id_belonging_to_user"), Version: utility.ToStringPtr("random_version_id")}
+			hasAccess, err := hasAnnotationPermission(ctx, &task, evergreen.AnnotationsView.Value)
+			assert.NoError(t, err)
+			assert.True(t, hasAccess)
+		},
+		"DoesNotOwnPatchAndDoesNotHaveAnnotationPermission": func(ctx context.Context, t *testing.T) {
+			task := restModel.APITask{ProjectId: utility.ToStringPtr("project_id_belonging_to_user"), Version: utility.ToStringPtr("random_version_id")}
+			hasAccess, err := hasAnnotationPermission(ctx, &task, evergreen.AnnotationsModify.Value)
+			assert.NoError(t, err)
+			assert.False(t, hasAccess)
+		},
+		"OwnsPatchButDoesNotHaveAnnotationPermission": func(ctx context.Context, t *testing.T) {
```

Comment:
it seems like the test name `DoesNotHaveAnnotationPermission` is misleading, because at the end of the function it seems to assert that the user does have view annotation permission